content/en/docs/tasks/administer-cluster/kubelet-config-file.md

```diff
@@ -73,6 +73,41 @@ If `--config` is provided and the values are not specified via the command line,
 defaults for the `KubeletConfiguration` version apply.
 In the above example, this version is `kubelet.config.k8s.io/v1beta1`.
 
+## Kubelet Drop-in Directory
+
+As of Kubernetes v1.28.0, the Kubelet has been extended to support a drop-in configuration directory. The location of it can be specified with
+`--config-dir` flag, and it defaults to `""`, or disabled, by default.
+
+When specified, the Kubelet must have the environment variable `KUBELET_CONFIG_DROPIN_DIR_ALPHA` set (the value does not matter).
+This approach was taken to allow users to opt-into an alpha feature, without a complicated reconcilliation process of feature gates between three different sources.
+If `--config-dir` is specified via CLI but `KUBELET_CONFIG_DROPIN_DIR_ALPHA` is not specified in the environment, the Kubelet will exit with an error.
+
+One can use the Kubelet configuration directory in a similar way to the Kubelet config file.
+**Note**: the suffix of a valid Kubelet drop-in configuration file must be `.conf`. For instance: `99-kubelet-address.conf`
+
+For instance, a user may want a baseline Kubelet configuration for all nodes, but may customize the `address` field. This can be done as follows:
+
+Kubelet Configuration:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
+kind: KubeletConfiguration
+port: 20250
+serializeImagePulls: false
+evictionHard:
+    memory.available:  "200Mi"
+```
+
+Contents of a file in `--config-dir` directory:
+```yaml
+apiVersion: kubelet.config.k8s.io/v1beta1
```

Comment:
Does this need to match. Eg
```diff
-apiVersion: kubelet.config.k8s.io/v1beta1
+apiVersion: kubelet.config.k8s.io/v1beta1 # can be different from main configuration
```

Reply:
I don't think necessarily though I haven't tested. We could add that to the things to check in the KEP and get back to it in beta?

Reply:
correct!